### PR TITLE
Draft: custom params set for execution mode

### PR DESCRIPTION
**Pull request by nkvch**

### CODE REVIEW

**Comment by nkvch** [comment]
Backend part is done. There are some minor frontend hazards that I will fix monday

File: backend/aiconsole/api/endpoints/execution_modes.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    if ExecutionModeParams is None:
+        _log.info(f"Module {module_name} does not have a ExecutionModeParams structure")
+        return None
+
+    if not issubclass(ExecutionModeParams, BaseModel):
+        _log.error(f"ExecutionModeParams in module {module_name} is not a subclass of BaseModel")
+        return None
+
+    return ExecutionModeParams.model_json_schema()["properties"]
```
Comment (udislau) [inline backend/aiconsole/api/endpoints/execution_modes.py:49]: this is a core logic, not api

File: backend/aiconsole/api/endpoints/execution_modes.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    if len(module_path.split(":")) != 2:
+        _log.error("Invalid module path format. Expected: 'module_name:object_name'")
+
+    module_name = module_path.split(":")[0]
+
+    try:
+        module = importlib.import_module(module_name)
+    except ModuleNotFoundError:
+        _log.error(f"Module {module_name} not found")
+        return None
```
Comment (udislau) [inline backend/aiconsole/api/endpoints/execution_modes.py:37]: User needs to see the err in the app

File: backend/aiconsole/core/assets/agents/agent.py [hunk 3]
```diff
@@ -23,4 +26,4 @@ class AICAgent(Asset):
     system: str
 
     gpt_mode: GPTMode = QUALITY_GPT_MODE
-    execution_mode: str = "aiconsole.core.chat.execution_modes.normal:execution_mode"
+    execution_mode: AgentExecutionMode = AgentExecutionMode()
```
Comment (udislau) [inline backend/aiconsole/core/assets/agents/agent.py:29]: Field(default_factory=...)

File: backend/aiconsole/core/assets/agents/agent_execution_modes/__init__.py
Comment (udislau) [inline backend/aiconsole/core/assets/agents/agent_execution_modes/__init__.py:1]: No need for now to create subdir, increases complexity when it's not needed

File: backend/aiconsole/core/chat/execution_modes/example_countdown.py [hunk 4]
```diff
@@ -41,12 +44,26 @@
 )
 
 
+class ExecutionModeParams(BaseModel):
+    from_: int = Field(default=10, alias="from")
```
Comment (udislau) [inline backend/aiconsole/core/chat/execution_modes/example_countdown.py:48]: What is "from_" and what  is alias "from" does?

File: backend/aiconsole/core/chat/execution_modes/example_countdown.py [hunk 5]
```diff
@@ -41,12 +44,26 @@
...
+    to: int = Field(default=1)
```
Comment (udislau) [inline backend/aiconsole/core/chat/execution_modes/example_countdown.py:49]: No need in Field(default ...) unless it is a callable

File: backend/aiconsole/core/chat/execution_modes/example_countdown.py [hunk 6]
```diff
@@ -41,12 +44,26 @@
...
+    message: str = Field(default="Hello world!")
+    should_notify: bool = Field(default=False)
+
+
 async def _execution_mode_process(
     chat_mutator: ChatMutator,
     agent: AICAgent,
     materials: list[AICMaterial],
     rendered_materials: list[RenderedMaterial],
+    **kwargs,
```
Comment (udislau) [inline backend/aiconsole/core/chat/execution_modes/example_countdown.py:59]: Don't use **kwargs in our code, is there a way to not use it here?

File: backend/aiconsole/preinstalled/agents/assistant.toml [hunk 7]
```diff
@@ -8,4 +8,4 @@ usage_examples = [
...
 gpt_mode = "quality"
-execution_mode = "aiconsole.core.chat.execution_modes.normal:execution_mode"
+execution_mode = { module_path = "aiconsole.core.chat.execution_modes.normal:execution_mode" }
```
Comment (udislau) [inline backend/aiconsole/preinstalled/agents/assistant.toml:11]: What is the purpose of {} for assignment?

**Review by udislau — CHANGES_REQUESTED** [review]

File: backend/aiconsole/api/endpoints/execution_modes.py (hunk 1 above)
Comment (nkvch) [reply backend/aiconsole/api/endpoints/execution_modes.py]: moved to core

File: backend/aiconsole/core/assets/agents/agent.py (hunk 3 above)
Comment (nkvch) [reply backend/aiconsole/core/assets/agents/agent.py]: done

File: backend/aiconsole/core/assets/agents/agent_execution_modes/__init__.py
Comment (nkvch) [reply backend/aiconsole/core/assets/agents/agent_execution_modes/__init__.py:1]: ok

File: backend/aiconsole/core/chat/execution_modes/example_countdown.py (hunk 4 above)
Comment (nkvch) [reply backend/aiconsole/core/chat/execution_modes/example_countdown.py]: that was generated by cursor, changed to "start"

File: backend/aiconsole/core/chat/execution_modes/example_countdown.py (hunk 6 above)
Comment (nkvch) [reply backend/aiconsole/core/chat/execution_modes/example_countdown.py]: replaced with params_values argument for this Protocol

File: backend/aiconsole/preinstalled/agents/assistant.toml (hunk 7 above)
Comment (nkvch) [reply backend/aiconsole/preinstalled/agents/assistant.toml:11]: agent's execution_mode is not a string anymore but a structure (dict) with module_path and params_values

**Review by udislau — APPROVED** [review]

**Pull request merged**